core/base/inc/Rtypes.h

```diff
@@ -284,98 +285,112 @@ private:          \
+   {                                                                                                            \
+      return ::ROOT::Internal::ClassDefGenerateInitInstanceLocalInjector<name>::Class();                        \
+   }                                                                                                            \
+   virtual_keyword void Streamer(TBuffer &R__b) overrd                                                          \
+   {                                                                                                            \
+      ::ROOT::Internal::DefaultStreamer(R__b, name::Class(), this);                                             \
+   }
 
 #define ClassDef(name,id)                            \
    _ClassDefOutline_(name,id,virtual,)               \
-   static int DeclFileLine() { return __LINE__; }
+   /** \cond HIDDEN_SYMBOLS */ static int DeclFileLine() { return __LINE__; } /** \endcond */
 
 #define ClassDefOverride(name,id)                    \
    _ClassDefOutline_(name,id,,override)              \
-   static int DeclFileLine() { return __LINE__; }
+   /** \cond HIDDEN_SYMBOLS */ static int DeclFileLine() { return __LINE__; } /** \endcond */
 
 #define ClassDefNV(name,id)                          \
    _ClassDefOutline_(name,id,,)                      \
-   static int DeclFileLine() { return __LINE__; }
+   /** \cond HIDDEN_SYMBOLS */ static int DeclFileLine() { return __LINE__; } /** \endcond */
 
 #define ClassDefInline(name,id)                      \
    _ClassDefInline_(name,id,virtual,)                \
-   static int DeclFileLine() { return __LINE__; }
+   /** \cond HIDDEN_SYMBOLS */ static int DeclFileLine() { return __LINE__; } /** \endcond */
 
 #define ClassDefInlineOverride(name,id)              \
    _ClassDefInline_(name,id,,override)               \
-   static int DeclFileLine() { return __LINE__; }
+   /** \cond HIDDEN_SYMBOLS */ static int DeclFileLine() { return __LINE__; } /** \endcond */
 
 #define ClassDefInlineNV(name,id)                    \
    _ClassDefInline_(name,id,,)                       \
-   static int DeclFileLine() { return __LINE__; }
+   /** \cond HIDDEN_SYMBOLS */ static int DeclFileLine() { return __LINE__; } /** \endcond */
 
 //#define _ClassDefInterp_(name,id) ClassDefInline(name,id)
 
 #define R__UseDummy(name) \
    class _NAME2_(name,_c) { public: _NAME2_(name,_c)() { if (name) { } } }
 
 
-#define ClassImpUnique(name,key) \
-   namespace ROOT { \
-      TGenericClassInfo *GenerateInitInstance(const name*); \
-      namespace { \
-         static int _R__UNIQUE_(_NAME2_(R__dummyint,key)) __attribute__((unused)) = \
-            GenerateInitInstance((name*)0x0)->SetImplFile(__FILE__, __LINE__); \
-         R__UseDummy(_R__UNIQUE_(_NAME2_(R__dummyint,key))); \
-      } \
+#define ClassImpUnique(name,key)                                                                        \
```

Comment:
`ClassImp` is deprecated and we could (if not already doing so) also tell doxygen to not expand and ignore it all-together.

Reply:
Hmm but it looks as if it is still being used in lots of places. 'grep ClassImp' in the repo returns 2000+ uses.

Reply:
Yes, but the information it provide it un-used (it fills the value of `GetImplFileName/Line`)

Reply:
Oh I see. But my point was, because we will see this in many places, maybe it's better to mark it as deprecated rather than hiding it completely? This way, if you hover on with the mouse or press F1, your IDE will tell you that directly.

Here an example on how it would look like:
![image](https://user-images.githubusercontent.com/10653970/150608177-4fc5b9af-14b5-4f52-a2cd-7dff54c14477.png)

Reply:
I agree with you.